spec/integration/indirector/node/json.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+require 'puppet/node'
+require 'puppet/indirector/node/json'
+
+describe Puppet::Node::Json do
+  describe '#save' do
+    subject(:indirection) { described_class.indirection }
+
+    let(:env) { Puppet::Node::Environment.create(:testing, []) }
+    let(:node) { Puppet::Node.new('node_name', :environment => env) }
+    let(:file) { File.join(Puppet[:client_datadir], "node", "node_name.json") }
+
+    before do
+      indirection.terminus_class = :json
+    end
+
+    it 'is saves node details' do
+      indirection.save(node)
+    end
```

Comment:
might be good to verify that the expected node data is written to `indirection.terminus.path(Puppet[:certname])`

Reply:
ah I see that's done below. Maybe just drop this ^ test.